From 15c12d0182ef55f7f26c3e4e4c235a319576ae48 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jiri@resnulli.us>
Date: Sun, 6 Jan 2013 12:41:57 +0000
Subject: [PATCH 481/858] ethtool: set addr_assign_type to NET_ADDR_SET when
 addr is passed on create

In case user passed address via netlink during create, NET_ADDR_PERM was set.
That is not correct so fix this by setting NET_ADDR_SET.

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 2afb9b533423a9b97f84181e773cf9361d98fed6)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 net/core/rtnetlink.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/net/core/rtnetlink.c b/net/core/rtnetlink.c
index 18c4847..919f0ab 100644
--- a/net/core/rtnetlink.c
+++ b/net/core/rtnetlink.c
@@ -1668,9 +1668,11 @@ struct net_device *rtnl_create_link(struct net *net,
 
  if (tb[IFLA_MTU])
   dev->mtu = nla_get_u32(tb[IFLA_MTU]);
- if (tb[IFLA_ADDRESS])
+ if (tb[IFLA_ADDRESS]) {
   memcpy(dev->dev_addr, nla_data(tb[IFLA_ADDRESS]),
     nla_len(tb[IFLA_ADDRESS]));
+  dev->addr_assign_type = NET_ADDR_SET;
+ }
  if (tb[IFLA_BROADCAST])
   memcpy(dev->broadcast, nla_data(tb[IFLA_BROADCAST]),
     nla_len(tb[IFLA_BROADCAST]));
-- 
1.8.3.1